app/products/calls/state/actions.ts

```diff
@@ -22,6 +22,18 @@ export const setCalls = (serverUrl: string, myUserId: string, calls: Dictionary<
     setChannelsWithCalls(serverUrl, channelsWithCalls);
 
     setCallsState(serverUrl, {serverUrl, myUserId, calls, enabled});
+
+    // Does the current call need to be updated?
+    const currentCall = getCurrentCall();
+    if (!currentCall || !calls[currentCall.channelId]) {
+        return;
+    }
+
+    const nextCall = {
+        ...currentCall,
+        ...calls[currentCall.channelId],
+    };
```

Comment:
Purely a technical question but do we need to spread the previous state? I'd expect the new one to be complete but maybe not always?

Reply:
Yep we do need to -- the CurrentCall object has a few other fields that aren't in the Call object.